x-pack/plugins/lens/public/react_embeddable/expressions/expression_params.ts

```diff
@@ -0,0 +1,236 @@
+    activeVisualizationState,
+    activeDatasourceState,
+  };
+}
+
+function buildHasCompatibleActions(api: LensApi, { uiActions }: LensEmbeddableStartServices) {
+  return async (event: ExpressionRendererEvent): Promise<boolean> => {
+    if (!uiActions?.getTriggerCompatibleActions) {
+      return false;
+    }
+    if (
+      isLensTableRowContextMenuClickEvent(event) ||
+      isLensMultiFilterEvent(event) ||
+      isLensFilterEvent(event)
+    ) {
+      const actions = await uiActions.getTriggerCompatibleActions(
+        VIS_EVENT_TO_TRIGGER[event.name],
+        {
+          data: event.data,
+          embeddable: api,
+        }
+      );
+
+      return actions.length > 0;
+    }
+
+    return false;
+  };
+}
+
+function buildGetCompatibleCellValueActions(
+  api: LensApi,
+  { uiActions }: LensEmbeddableStartServices
+): GetCompatibleCellValueActions {
+  return async (data) => {
+    if (!uiActions?.getTriggerCompatibleActions) {
+      return [];
+    }
+    const actions: Array<Action<CellValueContext>> = (await uiActions.getTriggerCompatibleActions(
+      CELL_VALUE_TRIGGER,
+      { data, embeddable: api }
+    )) as Array<Action<CellValueContext>>;
+    return actions
+      .sort((a, b) => (a.order ?? Infinity) - (b.order ?? Infinity))
+      .map((action) => ({
+        id: action.id,
+        type: action.type,
+        iconType: action.getIconType({ embeddable: api, data, trigger: cellValueTrigger })!,
+        displayName: action.getDisplayName({ embeddable: api, data, trigger: cellValueTrigger }),
+        execute: (cellData) =>
+          action.execute({ embeddable: api, data: cellData, trigger: cellValueTrigger }),
+      }));
+  };
+}
+
+export async function getExpressionRendererParams(
+  state: LensRuntimeState,
+  {
+    settings: { syncColors = true, syncCursor = true, syncTooltips = false },
+    services,
+    disableTriggers = false,
+    getExecutionContext,
+    searchSessionId,
+    abortController,
+    onRender,
+    handleEvent,
+    onData = noop,
+    logError,
+    api,
+    addUserMessages,
+    updateBlockingErrors,
+    searchContext,
+    renderCount,
+  }: GetExpressionRendererPropsParams
+): Promise<{
+  params: ExpressionWrapperProps | null;
+  abortController?: AbortController;
+  indexPatterns: IndexPatternMap;
+  indexPatternRefs: IndexPatternRef[];
+  activeVisualizationState?: unknown;
+  activeDatasourceState?: unknown;
+}> {
+  const { expressionRenderer, documentToExpression } = services;
+
+  const {
+    expression,
+    indexPatterns,
+    indexPatternRefs,
+    activeVisualizationState,
+    activeDatasourceState,
+  } = await getExpressionFromDocument(state.attributes, documentToExpression);
+
+  // if at least one indexPattern is time based, then the Lens embeddable requires the timeRange prop
+  // this is necessary for the dataview embeddable but not the ES|QL one
+  if (
+    // restore the legacy behavior for now to not break things
+    renderCount < 2 &&
```

Comment:
This continuos check was producing more and more errors on the solutions side.
I've restored the previous behaviour where the check was performed only for the first rendering, then skipped.
I've created a follow up issue here: https://github.com/elastic/kibana/issues/200236